### PR TITLE
feat(common-error-conditions): Add incomplete-customer-details test

### LINKED ISSUE #165: common-error-conditions - incomplete-customer-details - Expect an IncompleteCustomerDetailsError when customer details are incomplete
**Feature**

Tests C1, C2 and B for common error conditions applicable to all implementations

**Use Case**

Run each of C2 and B for a valid opportunity, with customer details incomplete, expecting an IncompleteCustomerDetailsError to be returned.

(C1 is ignored because customer details are not accepted for C1)

In the Booking API spec: https://openactive.io/open-booking-api/EditorsDraft/#order-creation-orderquote-orderproposal-and-order-error-responses

### PR DESCRIPTION
**Pull request by lukehesluke**
closes #165

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -5,30 +5,14 @@
     "version": "0.2.0",
     "configurations": [
         {
-            "name": "integration-tests - core/availability-check/opportunity-full",
+            "name": "integration-tests - core/common-error-conditions/incomplete-customer-details",
```
Comment (lukehesluke) [inline .vscode/launch.json:8]: Constantly editing this configuration is not ideal. Ideally, we'd have a "Integration Test this file", which is possible with `${file}` (https://github.com/microsoft/vscode-recipes/tree/master/debugging-mocha-tests#configure-launchjson-file-for-your-test-framework) but that wouldn't work here as it'd give the full path including `packages/openactive-integration-tests`.

Not very important though

File: packages/openactive-integration-tests/test/features/core/common-error-conditions/implemented/incomplete-customer-details-test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+   * Fetch some opportunities and run C1
+   *
+   * Note: This generates jest blocks like `beforeAll()`, `it()`, etc. Therefore, this must be run within a `describe()` block
+   *
+   * @param {RequestState} state
+   * @param {FlowHelper} state
+   */
+  function runC1(state, flow) {
+    beforeAll(async () => {
+      await state.fetchOpportunities(orderItemCriteria);
```
Comment (lukehesluke) [inline packages/openactive-integration-tests/test/features/core/common-error-conditions/implemented/incomplete-customer-details-test.js:34]: I've experimented by not including the `await chakram.wait()` line, which I don't think is needed as we await chakram requests (as far as I'm aware) as soon as they happen. Seems to work 👍

File: .vscode/launch.json (hunk 1 above)
Comment (nickevansuk) [reply .vscode/launch.json:8]: I think Jest can handle the full path?

File: packages/openactive-integration-tests/test/features/core/common-error-conditions/implemented/incomplete-customer-details-test.js (hunk 2 above)
Comment (nickevansuk) [reply packages/openactive-integration-tests/test/features/core/common-error-conditions/implemented/incomplete-customer-details-test.js:34]: Ye think this works if only waiting for one request...

**Pull request merged**